### PR TITLE
Add a field `rawHubURLPath` in the resource api

### PR DESCRIPTION
**Pull request by pratap0007**
Add a field rawHubURLPath in the resource api
    
This patch adds a field rawHubURLPath, which will contains
api url path to get the resource raw yaml file and this field
will be added in the following api's
    
    - /v1/resources
    - /v1/resource/<resource_id>
    - /v1/resource/<resource_id>/versions
    - /v1/resource/<catalog_name>/<resource_kind>/<resource_name>/version/<version>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
This will add a field `resourceURLPath` in the resource api and which can be is used to get the resource raw yaml file 
```

### CODE REVIEW

**Review by vinamra28 — APPROVED** [review]

File: api/design/types/type.go [hunk 1]
```diff
@@ -140,6 +140,9 @@ var ResourceVersionData = ResultType("application/vnd.hub.resource.version.data"
 		Format(FormatURI)
 		Example("webURL", "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.1/buildah.yaml")
 	})
+	Attribute("hubRawURLPath", String, "Api url path to get the resource raw yaml", func() {
+		Example("hubRawURLPath", "/v1/tekton/task/buildah/raw")
```
Comment (piyush-garg) [inline api/design/types/type.go:144]: we should remove `v1`

also, can you improve the message a bit - `path of the api to get the raw of resource from hub apiserver`

File: api/design/types/type.go [hunk 2]
```diff
@@ -244,6 +250,9 @@ var ResourceData = ResultType("application/vnd.hub.resource.data", "ResourceData
 	Attribute("hubURLPath", String, "Url path of the resource in hub", func() {
 		Example("hubURLPath", "tekton/task/buildah")
 	})
+	Attribute("hubRawURLPath", String, "Api url path to get the resource raw yaml", func() {
```
Comment (piyush-garg) [inline api/design/types/type.go:253]: we should remove `v1`

also, can you improve the message a bit - `path of the api to get the raw yaml of resource from hub apiserver`

File: api/design/types/type.go [hunk 3]
```diff
@@ -255,6 +264,7 @@ var ResourceData = ResultType("application/vnd.hub.resource.data", "ResourceData
 				"minPipelinesVersion": "0.12.1",
 				"rawURL":              "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.1/buildah.yaml",
 				"webURL":              "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.1/buildah.yaml",
+				"hubRawURLPath":       "/v1/tekton/task/buildah/latest/raw",
```
Comment (piyush-garg) [inline api/design/types/type.go:267]: remove v1

File: api/design/types/type.go [hunk 4]
```diff
@@ -342,25 +354,28 @@ var Versions = ResultType("application/vnd.hub.versions", "Versions", func() {
...
 			Value(Val{
-				"id":      2,
-				"version": "0.2",
-				"rawURL":  "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.2/buildah.yaml",
-				"webURL":  "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.2/buildah.yaml",
+				"id":            2,
+				"version":       "0.2",
+				"rawURL":        "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.2/buildah.yaml",
+				"webURL":        "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.2/buildah.yaml",
+				"hubRawURLPath": "/v1/resource/tektoncd/task/buildah/raw",
```
Comment (piyush-garg) [inline api/design/types/type.go:361]: it should not contain `v1`

File: api/design/types/type.go [hunk 5]
```diff
@@ -342,25 +354,28 @@ var Versions = ResultType("application/vnd.hub.versions", "Versions", func() {
...
 			Value([]Val{{
-				"id":      1,
-				"version": "0.1",
-				"rawURL":  "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.1/buildah.yaml",
-				"webURL":  "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.1/buildah.yaml",
+				"id":            1,
+				"version":       "0.1",
+				"rawURL":        "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.1/buildah.yaml",
+				"webURL":        "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.1/buildah.yaml",
+				"hubRawURLPath": "/v1/resource/tektoncd/task/buildah/0.1/raw",
```
Comment (piyush-garg) [inline api/design/types/type.go:372]: it should not contain `v1`

File: api/design/types/type.go [hunk 6]
```diff
@@ -342,25 +354,28 @@ var Versions = ResultType("application/vnd.hub.versions", "Versions", func() {
...
 			}, {
-				"id":      2,
-				"version": "0.2",
-				"rawURL":  "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.2/buildah.yaml",
-				"webURL":  "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.2/buildah.yaml",
+				"id":            2,
+				"version":       "0.2",
+				"rawURL":        "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.2/buildah.yaml",
+				"webURL":        "https://github.com/tektoncd/catalog/blob/main/task/buildah/0.2/buildah.yaml",
+				"hubRawURLPath": "/v1/resource/tektoncd/task/buildah/0.2/raw",
```
Comment (piyush-garg) [inline api/design/types/type.go:378]: it should not contain `v1`

File: api/gen/http/openapi.yaml [hunk 7]
```diff
@@ -1812,6 +1812,10 @@ definitions:
                 example:
                     - id: 1
                       name: image-build
+            hubRawURLPath:
+                type: string
+                description: Api url path to get the resource raw yaml
+                example: /v1/tekton/task/buildah/raw
```
Comment (piyush-garg) [inline api/gen/http/openapi.yaml:1818]: it should not contain `v1`

File: api/gen/http/openapi.yaml [hunk 8]
```diff
@@ -1861,12 +1865,14 @@ definitions:
             categories:
                 - id: 1
                   name: image-build
+            hubRawURLPath: /v1/tekton/task/buildah/raw
```
Comment (piyush-garg) [inline api/gen/http/openapi.yaml:1868]: it should not contain `v1`

File: api/gen/http/openapi.yaml [hunk 9]
```diff
@@ -1861,12 +1865,14 @@ definitions:
...
             hubURLPath: tekton/task/buildah
             id: 1
             kind: task
             latestVersion:
                 description: Buildah task builds source into a container image and then pushes it to a container registry.
                 displayName: Buildah
+                hubRawURLPath: /v1/tekton/task/buildah/latest/raw
```
Comment (piyush-garg) [inline api/gen/http/openapi.yaml:1875]: it should not contain `v1`

File: api/v1/service/resource/resource.go [hunk 10]
```diff
@@ -332,6 +332,7 @@ func initResource(r model.Resource) *resource.ResourceData {
 	}
 	res.Kind = r.Kind
 	res.HubURLPath = fmt.Sprintf("%s/%s/%s", r.Catalog.Name, r.Kind, r.Name)
+	res.HubRawURLPath = fmt.Sprintf("/v1/resource/%s/%s/%s/raw", r.Catalog.Name, strings.ToLower(r.Kind), r.Name)
```
Comment (piyush-garg) [inline api/v1/service/resource/resource.go:335]: remove `v1`

File: api/v1/service/resource/resource.go [hunk 11]
```diff
@@ -353,6 +354,7 @@ func initResource(r model.Resource) *resource.ResourceData {
 		WebURL:              lv.URL,
 		RawURL:              getStringReplacer(lv.URL, r.Catalog.Provider).Replace(lv.URL),
 		HubURLPath:          fmt.Sprintf("%s/%s/%s/%s", r.Catalog.Name, r.Kind, r.Name, lv.Version),
+		HubRawURLPath:       fmt.Sprintf("/resource/%s/%s/%s/raw", r.Catalog.Name, strings.ToLower(r.Kind), r.Name),
```
Comment (PuneetPunamiya) [inline api/v1/service/resource/resource.go:357]: I think there should be version in the latestVersion struct 
i.e. resource/tekton/task/git-clone/0.9/raw

Comment (PuneetPunamiya) [inline api/v1/service/resource/resource.go:357]: Also we don't need a `/` before `resource`

**Pull request merged**